File: Volume0/Vol0027.c
#include <stdio.h>

int main(void)
{
    int m, d;
    int days[11] = { 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30 };
    char day[7][10] = { "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday" };

    while (scanf("%d %d", &m, &d), m != 0)
    {
        while (--m > 0) d += days[m - 1];

        printf("%s\n", day[(d + 2) % 7]);
    }

    return 0;
}

File: Sources/Vol00/Vol0027.hs
import Control.Applicative
import Control.Monad (when)


days :: Int -> Int -> Int

days m d = d + sum (take (m - 1) xs)
  where
    xs = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30]


solve :: (Int, Int) -> String

solve (m, d) = last $ take (days m d) xs
  where
    xs = cycle ["Thursday", "Friday", "Saturday", "Sunday", "Monday", "Tuesday", "Wednesday"]


input :: String -> (Int, Int)

input s = (x, y)
  where
    [x, y] = map read $ words s


main :: IO ()

main = do
  (m, d) <- input <$> getLine

  when (m /= 0) $ do
    putStrLn $ solve (m, d)

    main